app/controllers/search_controller.rb

```diff
@@ -0,0 +1,80 @@
+require 'uri'
+require 'net/http'
+require 'openssl'
+
+
+class SearchController < ApplicationController
+
+    before_action :init_results
+
+    def index
+        
+
+    end
+
+    def init_results
+        @resultsName = []
+    end
+
+    def show
+        
+        #search by title name
+        @titleName = params[:search][:title]
+        id_url = URI("https://watchmode.p.rapidapi.com/search/?search_field=name&search_value=#@titleName")
+        puts "title: is "
+        puts params[:search][:title]
+        puts params
+        
+
+        http = Net::HTTP.new(id_url.host, id_url.port)
+        http.use_ssl = true
+        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+        
+        idrequest = Net::HTTP::Get.new(id_url)
+        idrequest["x-rapidapi-host"] = 'watchmode.p.rapidapi.com'
+        idrequest["x-rapidapi-key"] = '3f2560c7e3msh18d24d47fcda8f8p17b513jsn0a997cc33ab8'
+        
+        @idresponse = http.request(idrequest)
+        @idresult = JSON.parse(@idresponse.read_body)
+        
+        i=0
+        #while loop for multiple shows with the same name (use riverdale as an example)
+        
+        @resultsURL = []
+        @services_result
+        while i < @idresult.size
+            @id = @idresult['title_results'][i]['id']
```

Comment:
needs null check. i got this error 
![image](https://user-images.githubusercontent.com/43143866/159810922-a07a5c6a-0573-4257-aa5a-3c04bf8fc1bc.png)